TraversabilityGrid: Fixes and CleanUp

JanWehrmann opened the pull request:
Fixed some Issues with TraversabilityGrid and TraversabilityClass, added unit tests for these Classes and made some whitespace changes as well as cleaned up some parts of the code.

Commit:
Whitespace changes only.

Removed trailing spaces and fixed indentation

Commit:
Cleanup.

Cleaned up some parameter names and types.

saarnold commented:
Checking the limits for the probability I understand but why changing the class ID from uint8 to int and then checking every time if it's in [0,255]? This is guaranteed when you use uint8. Can you explain this decision?

Commit:
Unit tests for TraversabilityGrid.

Includes test cases for TraversabilityCell and
TraversabilityClass.

saarnold commented on src/grid/TraversabilityGrid.cpp:
```diff
@@ -66,11 +69,22 @@ const TraversabilityClass& TraversabilityGrid::getTraversability(size_t x, size_
-void TraversabilityGrid::setProbability(float probability, size_t x, size_t y)
+uint8_t TraversabilityGrid::getTraversabilityClassId(std::size_t x, std::size_t y) const
 {
     TraversabilityCell cell = this->GridMap::at(x, y);
```
This should be a const reference, in order to avoid that a copy of the cell is done.

Commit:
TraversabilityGrid fixes.

Fixed multiple issues is TraversabilityGrid and TraversabilityClass
and overhauled the following methods:
TraversabilityGrid
- setTraversabilityAndProbability and setProbability
	These methods now return bool instead of void. Used to constrict
	probability values to [0, 1].
- registerNewTraversabilityClass
	Now correctly registers to the next available ID instead of
	skipping	every second ID.
TraversabilityClass
- isSet
	The Method was renamed to isClassDefined and now behaves	as
	expected.

saarnold commented:
Ok thanks!
